projects/igniteui-angular/src/lib/grid/column.component.ts

```diff
@@ -161,80 +381,205 @@ export class IgxColumnComponent implements AfterContentInit {
         return this._defaultMinWidth;
     }
-
-    get grid(): IgxGridComponent {
+/**
+ * Returns reference to the `igx-grid`.
+ * ```typescript
+ * let gridComponent = this.column.grid;
+ * ```
+ * @memberof IgxColumnComponent
+ */
+get grid(): IgxGridComponent {
         return this.gridAPI.get(this.gridID);
     }
-
-    get bodyTemplate(): TemplateRef<any> {
+/**
+ * Returns a reference to the `bodyTemplate`.
+ * ```typescript
+ * let bodyTemplate = this.column.bodyTemplate;
+ * ```
+ * @memberof IgxColumnComponent
+ */
+get bodyTemplate(): TemplateRef<any> {
         return this._bodyTemplate;
     }
-
-    set bodyTemplate(template: TemplateRef<any>) {
+/**
+ * Sets the body template.
+ * ```typescript
+ * @ViewChild("'bodyTemplate'", {read: TemplateRef })
+ * public bodyTemplate: TemplateRef<any>;
+ * this.column.bodyTemplate(bodyTemplate);
+ * ```
+ * @memberof IgxColumnComponent
+ */
+set bodyTemplate(template: TemplateRef<any>) {
         this._bodyTemplate = template;
         this.grid.markForCheck();
     }
-
-    get headerTemplate(): TemplateRef<any> {
+/**
+ * Returns a reference to the header template.
+ * ```typescript
+ * let headerTemplate = this.column.headerTemplate;
+ * ```
+ * @memberof IgxColumnComponent
+ */
+get headerTemplate(): TemplateRef<any> {
         return this._headerTemplate;
     }
-
-    set headerTemplate(template: TemplateRef<any>) {
+/**
+ * Sets the header template.
+ * ```typescript
+ * @ViewChild("'headerTemplate'", {read: TemplateRef })
+ * public headerTemplate: TemplateRef<any>;
+ * this.column.headerTemplate(headerTemplate);
+ * ```
+ * @memberof IgxColumnComponent
+ */
+set headerTemplate(template: TemplateRef<any>) {
         this._headerTemplate = template;
         this.grid.markForCheck();
     }
-
-    get footerTemplate(): TemplateRef<any> {
+/**
+ * Returns a reference to the footer template.
+ * ```typescript
+ * let footerTemplate = this.column.footerTemplate;
+ * ```
+ * @memberof IgxColumnComponent
+ */
+get footerTemplate(): TemplateRef<any> {
         return this._headerTemplate;
     }
-
-    set footerTemplate(template: TemplateRef<any>) {
+/**
+ * Sets the footer template.
+ * ```typescript
+ * @ViewChild("'footerTemplate'", {read: TemplateRef })
+ * public footerTemplate: TemplateRef<any>;
+ * this.column.footerTemplate(footerTemplate);
+ * ```
+ * @memberof IgxColumnComponent
+ */
+set footerTemplate(template: TemplateRef<any>) {
         this._footerTemplate = template;
         this.grid.markForCheck();
     }
-
-    get inlineEditorTemplate(): TemplateRef<any> {
+/**
+ * Returns a reference to the inline editor template.
+ * ```typescript
+ * let inlineEditorTemplate = this.column.inlineEditorTemplate;
+ * ```
+ * @memberof IgxColumnComponent
+ */
+get inlineEditorTemplate(): TemplateRef<any> {
         return this._inlineEditorTemplate;
     }
-
-    set inlineEditorTemplate(template: TemplateRef<any>) {
+/**
+ * Sets the inline editor template.
```

Comment:
Provide fully functional html template in the example, so that a dev can copy paste it and it works.